### ISSUE TITLE
[Bug]: Adnroid Emulator on first start hasInternetAccess returns false

### ISSUE THREAD

**Issue by umair9430** [OutdatedGuy/internet_connection_checker_plus#121]
### Is there an existing issue for this?

- [x] I have searched the existing issues

### Bug Description

When running the app on Android Emulator immediately after first install / cold start, my connectivity check (hasInternetAccess / isConnected) returns false, even though:
	•	WebSocket connects successfully
	•	Firebase initializes successfully
	•	Actual internet is clearly available

This behavior only occurs on Android Emulator.

On other platforms:
	•	✅ Real Android device → works correctly
	•	✅ iOS Simulator → works correctlysteps

### Steps to Reproduce

	1.	Cold boot Android Emulator
	2.	Install the app (fresh install)
	3.	Launch the app
	4.	Wait for app initialization
	•	Firebase connects
	•	WebSocket connects
	5.	Trigger connectivity check (hasInternetAccess / isConnected)
	6.	Observe returned value

### Expected Behavior

When the device/emulator has an active internet connection:
	•	hasInternetAccess / isConnected should return true
	•	Especially when other network services (WebSocket, Firebase) are already connected
	•	Behavior should be consistent across platforms:
	•	Android Emulator
	•	Real Android devices
	•	iOS Simulator

The first connectivity check after app install or cold start should not produce a false negative.

### Actual Behavior

On Android Emulator (first app launch after fresh install or cold boot):
	•	hasInternetAccess / isConnected returns false
	•	At the same time:
	•	WebSocket connects successfully
	•	Firebase initializes successfully
	•	Retrying the same connectivity check or restarting the app returns true

This issue does not occur on real Android devices or iOS Simulator.

### Package Version

2.9.1+2

### Platform(s)

Android

### Flutter Channel

stable

### Flutter Doctor Output

```shell
[✓] Flutter (Channel stable, 3.38.3, on macOS 15.3 24D60 darwin-arm64, locale en-PT) [611ms]
    • Flutter version 3.38.3 on channel stable at /Users/umair.k/Developer/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision 19074d12f7 (2 months ago), 2025-11-20 17:53:13 -0500
    • Engine revision 13e658725d
    • Dart version 3.10.1
    • DevTools version 2.51.1
    • Feature flags: enable-web, enable-linux-desktop, enable-macos-desktop, enable-windows-desktop, enable-android, enable-ios, cli-animations, enable-native-assets, omit-legacy-version-file, enable-lldb-debugging

[✓] Android toolchain - develop for Android devices (Android SDK version 36.0.0-rc4) [1,590ms]
    • Android SDK at /Users/umair.k/Library/Android/sdk
    • Emulator version 35.3.11.0 (build_id 12836668) (CL:N/A)
    • Platform android-36, build-tools 36.0.0-rc4
    • Java binary at: /Applications/Android Studio.app/Contents/jbr/Contents/Home/bin/java
      This is the JDK bundled with the latest Android Studio installation on this machine.
      To manually set the JDK path, use: `flutter config --jdk-dir="path/to/jdk"`.
    • Java version OpenJDK Runtime Environment (build 21.0.4+-12422083-b607.1)
    • All Android licenses accepted.
```

### Minimal Reproduction Code

```dart
sealed class InternetConnectivityGuard {
  Future<void> ensureConnectedToInternet();
}

class InternetConnectivityGuardImpl implements InternetConnectivityGuard {
  InternetConnectivityGuardImpl({
    required NetworkInfoService networkInfo,
  }) : _networkInfo = networkInfo;

  final NetworkInfoService _networkInfo;

  @override
  Future<void> ensureConnectedToInternet() async {
    final bool isConnected = await _networkInfo.isConnected;
    if (!isConnected) {
      final String message =
          getItInstance.isRegistered<InternetFailureMessage>()
              ? getItInstance<InternetFailureMessage>().text
              : _Constants.noInternetConnectionMessage;
      throw NoInternetException(
        message: message,
      );
    }
  }
}

sealed class NetworkInfoService {
  Future<bool> get isConnected;
}

class NetworkInfoServiceImpl implements NetworkInfoService {
  NetworkInfoServiceImpl({
    required Connectivity connectivity,
    required InternetConnection checker,
  })  : _connectivity = connectivity,
        _checker = checker;

  final Connectivity _connectivity;
  final InternetConnection _checker;

  @override
  Future<bool> get isConnected async {
    final List<ConnectivityResult> result =
        await _connectivity.checkConnectivity();

    final bool hasNetwork = result.contains(ConnectivityResult.wifi) ||
        result.contains(ConnectivityResult.mobile) ||
        result.contains(ConnectivityResult.vpn);
    if (!hasNetwork) {
      return false;
    }
    return _checker.hasInternetAccess;
  }
}
```

### Relevant Log Output

```shell

```

### Device/Network Information

```markdown

```

### Would you like to work on this issue?

- [ ] I'm interested in working on this issue

**Comment by OutdatedGuy** [comment]
Hi @umair9430, please check if this issue still persists in [v3](https://pub.dev/packages/internet_connection_checker_plus/versions/3.0.0).

If it persists, check if same is reproducible using example app of this package.

**Comment by khaledsawan** [comment]
Same for me, even on v3, on Android Emulator 

I also get an error on the iOS simulator with 
An exception has occurred.
TimeoutException (TimeoutException after 0:00:03.000000: Future not completed)

<img width="1010" height="740" alt="Image" src="https://github.com/user-attachments/assets/4000fb47-a282-4def-9a9c-9349de09027a" />

**Comment by OutdatedGuy** [comment]
@khaledsawan hmmmm, that's weird, as the `Exception` should be caught by the catch block at line 193.

Gemini gave me below response, can you check once if that fixes the issue?

### For VS Code
In your project's `.vscode/settings.json` file, you can explicitly disable debugging for external libraries to prevent the debugger from stepping into third-party code:

```json
{
    "dart.debugSdkLibraries": false,
    "dart.debugExternalLibraries": false
}
```

or

Check if the `All Exceptions` checkbox is ticked and untick it in Debug View
<img width="292" height="68" alt="Image" src="https://github.com/user-attachments/assets/79ba8326-90f4-45b6-9406-845e6486d9fa" />

**Comment by khaledsawan** [comment]
@OutdatedGuy I apologize for the delayed response. Yes, regarding the "Packages Debug" setting, I have unchecked the corresponding box.

However—returning to the original issue—I believe it still persists.
I would also like to point out that during the first 500 milliseconds of the application's runtime, the internet connection is reported as unavailable (returning a value of "False").

**Comment by OutdatedGuy** [comment]
@khaledsawan can you verify once if the same issue is reproducible on this package's example app? If yes, please provide debug logs of that run and event.

**Comment by OutdatedGuy** [comment]
Closing as stale. Feel free to comment to reopen this issue if it still persists.
